pages/_app.js

```diff
@@ -32,20 +33,19 @@ const fonts = [
   },
 ];
 
-class Layout extends React.Component {
-  render() {
-    // eslint-disable-next-line react/prop-types
-    const { children } = this.props;
+Layout.propTypes = {
+  children: PropTypes.node.isRequired,
+};
 
-    return (
-      <>
-        <Nav />
-        <main>{children}</main>
-        <Footer />
-        <ScrollUpButton />
-      </>
-    );
-  }
+function Layout({ children }) {
+  return (
+    <>
+      <Nav />
+      <main>{children}</main>
+      <Footer />
+      <ScrollUpButton />
+    </>
+  );
```

Comment:
I was unsure on doing the OperationCodeApp component since it extends App. I don't think anything from next can be a functional component.
https://spectrum.chat/next-js/general/is-there-an-example-of-app-js-as-a-functional-component~bfc80353-edf0-4850-9442-5d6c2b030209?m=MTU0NzU1MDA0NDUwNQ==

Reply:
Good to know!